File: backend-api/internal/models/map.go
package models

type ROMMap struct {
	ID           string  `json:"id"`
	DetailURL    string  `json:"detail_url"`
	Image        *string `json:"image"`
	Name         string  `json:"name"`
	MonsterCount int     `json:"monster_count"`
}

type ROMMapMonster struct {
	ID            int     `json:"id"`
	MapID         string  `json:"map_id"`
	MonsterID     *string `json:"monster_id"`
	MonsterName   *string `json:"monster_name"`
	MonsterImage  *string `json:"monster_image"`
	MonsterURL    *string `json:"monster_url"`
	Level         *string `json:"level"`
	Race          *string `json:"race"`
	Element       *string `json:"element"`
	Size          *string `json:"size"`
	RelationIndex int     `json:"relation_index"`
}

type ROMMapDetail struct {
	ID        string          `json:"id"`
	DetailURL string          `json:"detail_url"`
	Image     *string         `json:"image"`
	Name      string          `json:"name"`
	RawHTML   *string         `json:"raw_html"`
	Monsters  []ROMMapMonster `json:"monsters"`
}

File: archive-romhandbook-ui/lib/types/Map.ts
export type ROMMap = {
    id: string
    detail_url: string
    image: string | null
    name: string
    monster_count: number
}

export type ROMMapMonster = {
    id: number
    map_id: string
    monster_id: string | null
    monster_name: string | null
    monster_image: string | null
    monster_url: string | null
    level: string | null
    race: string | null
    element: string | null
    size: string | null
    relation_index: number
}

export type ROMMapDetail = {
    id: string
    detail_url: string
    image: string | null
    name: string
    raw_html: string | null
    monsters: ROMMapMonster[]
}

export type PaginatedMapResponse = {
    success: boolean
    data: ROMMap[]
    meta: {
        page: number
        limit: number
        total: number
        has_next: boolean
    }
}